Automattic/wp-calypso
# Components: Add layout component

#### Changes proposed in this Pull Request

With the recent [width updates](https://github.com/Automattic/wp-calypso/pull/46616) to our account section, this PR does some initial work to abstract a layout component from our `/hosting-config` screen. We'll be using this component to update some layouts in `/me` — most notably the purchase management setting screen.

![image](https://user-images.githubusercontent.com/6981253/97744074-fc0fcd80-1abc-11eb-9b56-d116417e4a81.png)


#### Testing instructions

* Start by reviewing `/hosting-config` to confirm nothing is broken there. 
* Review `/devdocs/design/layout` on multiple browsers to confirm is works consistently.
